pkg/traceql/engine.go

```diff
@@ -171,7 +171,15 @@ func (e *Engine) asTraceSearchMetadata(spanset *Spanset) (*tempopb.TraceSearchMe
 			Attributes:        nil,
 		}
 
+		if name, ok := span.Attributes[NewIntrinsic(IntrinsicName)]; ok {
+			tempopbSpan.Name = name.S
+		}
+
 		for attribute, static := range span.Attributes {
+			if attribute.Name == IntrinsicName.String() || attribute.Name == IntrinsicDuration.String() {
```

Comment:
This could be `attribute.Intrinsic == ...`

Reply:
Yeah, good catch!